sdk/tests/platform/test_az_mqtt5_policy.c

```diff
@@ -0,0 +1,400 @@
+          &test_mqtt5_property_bag,
+          AZ_MQTT5_PROPERTY_TYPE_CONTENT_TYPE,
+          &test_mqtt5_property_string),
+      AZ_OK);
+  assert_int_equal(
+      az_mqtt5_property_bag_stringpair_append(
+          &test_mqtt5_property_bag,
+          AZ_MQTT5_PROPERTY_TYPE_USER_PROPERTY,
+          &test_mqtt5_property_string_pair1),
+      AZ_OK);
+  assert_int_equal(
+      az_mqtt5_property_bag_stringpair_append(
+          &test_mqtt5_property_bag,
+          AZ_MQTT5_PROPERTY_TYPE_USER_PROPERTY,
+          &test_mqtt5_property_string_pair2),
+      AZ_OK);
+  assert_int_equal(
+      az_mqtt5_property_bag_int_append(
+          &test_mqtt5_property_bag,
+          AZ_MQTT5_PROPERTY_TYPE_MESSAGE_EXPIRY_INTERVAL,
+          TEST_MQTT_PROPERTY_MESSAGE_EXPIRY_INTERVAL),
+      AZ_OK);
+  assert_int_equal(
+      az_mqtt5_property_bag_byte_append(
+          &test_mqtt5_property_bag,
+          AZ_MQTT5_PROPERTY_TYPE_PAYLOAD_FORMAT_INDICATOR,
+          TEST_MQTT_PROPERTY_PAYLOAD_FORMAT_INDICATOR),
+      AZ_OK);
+  assert_int_equal(
+      az_mqtt5_property_bag_binary_append(
+          &test_mqtt5_property_bag,
+          AZ_MQTT5_PROPERTY_TYPE_CORRELATION_DATA,
+          &test_mqtt5_property_binary_data),
+      AZ_OK);
+
+  az_mqtt5_pub_data test_mqtt5_pub_data = {
+    .topic = AZ_SPAN_LITERAL_FROM_STR(TEST_MQTT_TOPIC),
+    .qos = 0,
+    .out_id = 0,
+    .payload = AZ_SPAN_LITERAL_FROM_STR(TEST_MQTT_PAYLOAD),
+    .properties = &test_mqtt5_property_bag,
+  };
+
+  assert_int_equal(az_mqtt5_outbound_pub(&test_mqtt5_client, &test_mqtt5_pub_data), AZ_OK);
+
+  int retries = TEST_MAX_RESPONSE_CHECKS;
+  while (ref_puback == 0 && ref_recv == 0 && retries > 0)
+  {
+    assert_int_equal(az_platform_sleep_msec(TEST_RESPONSE_DELAY_MS), AZ_OK);
+    retries--;
+  }
+
+  assert_int_equal(ref_puback, 1);
+  assert_true(ref_recv > 0);
+
+#ifdef TRANSPORT_MOSQUITTO
+  mosquitto_property_free_all(&test_mqtt5_property_bag._internal.options.properties);
+#endif
+}
+
+static void test_az_mqtt5_policy_outbound_disconnect_success(void** state)
+{
+  (void)state;
+  ref_disconnect = 0;
+
+  assert_int_equal(az_mqtt5_outbound_disconnect(&test_mqtt5_client), AZ_OK);
+
+  int retries = TEST_MAX_RESPONSE_CHECKS;
+  while (ref_disconnect == 0 && retries > 0)
+  {
+    assert_int_equal(az_platform_sleep_msec(TEST_RESPONSE_DELAY_MS), AZ_OK);
+    retries--;
+  }
+
+  assert_int_equal(ref_disconnect, 1);
+}
+
+int test_az_mqtt5_policy()
+{
+  const struct CMUnitTest tests[] = {
+#ifndef AZ_NO_PRECONDITION_CHECKING
+    cmocka_unit_test(test_az_mqtt5_policy_init_null_failure),
+#endif // AZ_NO_PRECONDITION_CHECKING
+    cmocka_unit_test(test_az_mqtt5_policy_init_success),
+    cmocka_unit_test(test_az_mqtt5_policy_init_valid_success),
+    cmocka_unit_test(test_az_mqtt5_policy_outbound_connect_success),
+    cmocka_unit_test(test_az_mqtt5_policy_outbound_sub_success),
+    cmocka_unit_test(test_az_mqtt5_policy_outbound_pub_properties_success),
```

Comment:
do we need a test without properties too?

Reply:
Added test w/o properties!